rudder-web/src/main/scala/com/normation/rudder/web/rest/directive/DirectiveAPIService2.scala

```diff
@@ -302,7 +302,7 @@ case class DirectiveAPIService2 (
     }
   }
 
-  def updateDirective(id: String, req: Req, restValues : Box[RestDirective]) = {
+  def updateDirective(id: String, req: Req, restValues : Box[RestDirective], check : Boolean) = {
```

Comment:
I would much, much more prefer to have two separated method here, because:

- boolean parameters are quite obscure when seen at call site, 
- the method has two really different behavior depending of the boolean value: one read only + check, the other modify stored value. 
- and even the implementation is completly different for the two sides of the else.